lib/public/AppFramework/App.php

```diff
@@ -95,7 +95,7 @@ public function __construct(string $appName, array $urlParams = []) {
 				}
 			}
 
-			if (!$setUpViaQuery) {
+			if (!$setUpViaQuery && $applicationClassName !== 'OCP\\AppFramework\\App') {
```

Comment:
```suggestion
			if (!$setUpViaQuery && $applicationClassName !== \OCP\AppFramework\App::class) {
```

php will expand this to the fully qualified class name

Reply:
Ah sure.